homeassistant/components/recorder/__init__.py

```diff
@@ -511,10 +513,11 @@ def _async_setup_periodic_tasks(self):
         async_track_time_change(
             self.hass, self.async_nightly_tasks, hour=4, minute=12, second=0
         )
-        # Compile hourly statistics every hour at *:12
-        async_track_time_change(
-            self.hass, self.async_hourly_statistics, minute=12, second=0
-        )
+        if not self.hass.data.get(DISABLE_STATISTICS):
```

Comment:
Why not just patch `self.async_hourly_statistics` out in the tests to be a no-op?

Reply:
OK, we could maybe patch the `Recorder` constructor to return an object where the method is optionally mocked out.

Reply:
Fixed in 2235474